stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerView.kt

```diff
@@ -1,56 +1,31 @@
 package io.getstream.chat.android.ui.message.composer
 
 import android.content.Context
+import android.text.Editable
 import android.util.AttributeSet
-import android.view.View
-import android.view.ViewGroup
+import android.widget.FrameLayout
 import androidx.constraintlayout.widget.ConstraintLayout
-import io.getstream.chat.android.client.models.Attachment
-import io.getstream.chat.android.common.state.MessageAction
+import androidx.core.view.children
+import androidx.core.view.isVisible
+import com.google.android.material.internal.TextWatcherAdapter
+import io.getstream.chat.android.common.state.MessageInputState
 import io.getstream.chat.android.core.ExperimentalStreamChatApi
 import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
 import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultCenterContentBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultLeadingContentBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultTrailingContentBinding
 
 @ExperimentalStreamChatApi
 public class MessageComposerView : ConstraintLayout {
 
     private lateinit var binding: StreamUiMessageComposerBinding
 
-    /**
-     * @property leadingContent A function returning content visible on the left.
-     */
-    public var leadingContent: MessageComposerView.(ViewGroup) -> View = { emptyView() }
-        set(value) {
-            field = value
-            binding.leadingContent.apply {
-                removeAllViews()
-                addView(value(this))
-            }
-        }
+    public var onSendMessage: () -> Unit = {}
```

Comment:
Should we expose a message as a parameter here, so that our users can potentially override it/add extra data?

Reply:
My understanding is this would enforce making `MessageComposerView` stateful. If `MessageComposerView` was responsible for instantiating `Message` it would need to store state internally 🤔

Reply:
IMO, this is something we can provide via `MessageComposerViewModel::bindView` and in case `MessageComposerView` is used without the default view model it's the responsibility of the external controller to hold the state and allow instantiating the `Message`